src/material/select/select.ts

```diff
@@ -677,6 +680,7 @@ export class MatSelect extends _MatSelectMixinBase implements AfterContentInit,
    */
   writeValue(value: any): void {
     if (this.options) {
```

Comment:
We're so close to getting this in, just one more change needs to be made I think. 

`writeValue` is called by the forms system before the content is initialized, which means that `this.options` will always be false when it is initially set. What if we change this to be simply

```
  writeValue(value: any): void {
    this.value = value;
  }
```

The value setter will not run `setSelectionByValue` unless there are options, but importantly it will preserve the passed `value` from forms. Then, when content is initialized, options are changed and then `_initializeSelection` will be called, which in turn calls `setSelectionByValue` with the current (and correct) value.

Reply:
Added.